src/components/StatusBadge/StatusBadge.tsx

```diff
@@ -1,12 +1,28 @@
 import unpaidIcon from '../../assets/images/paymentrequest_status_unpaid.svg';
 import paidIcon from '../../assets/images/paymentrequest_status_paid.svg';
 import partiallyPaidIcon from '../../assets/images/paymentrequest_status_partial.svg';
+import { cva } from 'class-variance-authority';
+
+const badge = cva('rounded-full py-1 px-2 space-x-2 inline-flex items-center', {
+  variants: {
+    intent: {
+      paid: ['bg-[#D8F2EA]', 'text-[#004D33]'],
+      partial: ['bg-[#F2EED8]', 'text-[#663300]'],
+      unpaid: ['bg-[#F1F3F4]', 'text-default-text'],
+    },
+  },
+  defaultVariants: {
+    intent: 'unpaid',
+  },
+});
+
+type Status = 'UNPAID' | 'PARTIAL' | 'PAID';
 
 interface StatusBadgeProps {
-  status: 'UNPAID' | 'PARTIAL' | 'PAID';
+  status: String;
 }
```

Comment:
I might not have made myself clear with the previous message regarding using upper or lower case and this could've generated a misunderstanding, apologies.

I was thinking in something more like receiving the props in lower case (as usual) and then make the necessary adjustments in the UI.

Here's kind of how I was thinking it:
```jsx
type Status = 'unpaid' | 'partial' | 'paid'; // 1. Options are in lowered case as usual

interface StatusBadgeProps {
  status: Status; // 2. Type is being used (instead of `string`) 
}

const getIconForStatus = (status: Status) => {
  switch (status) {
    case 'partial':
      return partiallyPaidIcon;
    case 'paid':
      return paidIcon;
    case 'unpaid':
    default:
      return unpaidIcon;
  }
};

// 3. Note that the `parseStatus` function won't do anything now because we already got the type from the prop itself (see point 2)

const StatusBadge = ({ status }: StatusBadgeProps) => {
  // 4. No conditional rendering is needed
  return (
    <div className={badge({ intent: status })}> 
      <img className="h-3 w-3" src={getIconForStatus(status)} alt={status} />
      <span className="text-xs font-normal uppercase">{status.toUpperCase()}</span> {/* 5. We are uppercasing the string ONLY for the UI */}
    </div>
  );
```

---

With this changes you then will have to update the story for this component,